samza-core/src/main/java/org/apache/samza/startpoint/StartpointSerde.java

```diff
@@ -30,21 +29,16 @@
 class StartpointSerde implements Serde<Startpoint> {
   private static final String STARTPOINT_CLASS = "startpointClass";
   private static final String STARTPOINT_OBJ = "startpointObj";
-  // Map required for the object mapper to know how to deserialize into the Startpoint subclass
-  private static final Map<String, Class<? extends Startpoint>> SERDE_TYPES = ImmutableMap.of(
-      StartpointSpecific.class.getCanonicalName(), StartpointSpecific.class,
-      StartpointTimestamp.class.getCanonicalName(), StartpointTimestamp.class,
-      StartpointEarliest.class.getCanonicalName(), StartpointEarliest.class,
-      StartpointLatest.class.getCanonicalName(), StartpointLatest.class,
-      StartpointBootstrap.class.getCanonicalName(), StartpointBootstrap.class);
 
   private final ObjectMapper mapper = new ObjectMapper();
 
   @Override
   public Startpoint fromBytes(byte[] bytes) {
     try {
       LinkedHashMap<String, String> deserialized = mapper.readValue(bytes, LinkedHashMap.class);
-      return mapper.readValue(deserialized.get(STARTPOINT_OBJ), SERDE_TYPES.get(deserialized.get(STARTPOINT_CLASS)));
+      Class<? extends Startpoint> startpointClass =
+          (Class<? extends Startpoint>) Class.forName(deserialized.get(STARTPOINT_CLASS));
```

Comment:
Would it be useful to check if startpointClass extends Startpoint before casting? Not sure how worried we should be that something else unexpectedly writes to this store.

Reply:
Might not be necessary to check, but this should throw a cast exception if it's not a `Startpoint` subclass and at least provide a small layer of safety.

Reply:
Ok, if it throws a class cast exception at this line, then you should be good.
I thought in some cases, incorrect casts might reveal themselves later, and I wasn't sure how far the code would go here before failing (if at all).